PluginGenerator/PluginBuilder.cs

```diff
@@ -160,8 +210,101 @@ public void Build()
+
+            if (string.IsNullOrWhiteSpace(this.FindPluginKey()))
+            {
+                throw new InvalidOperationException(UIResources.CoreBuilder_PluginKeyIsRequired);
+            }
+
+            if (string.IsNullOrWhiteSpace(this.FindPluginName()))
+            {
+                throw new InvalidOperationException(UIResources.CoreBuilder_PluginNameIsRequired);
+            }
+        }
+
+        private string FindPluginKey()
+        {
+            string pluginKey;
+            this.properties.TryGetValue(WellKnownPluginProperties.Key, out pluginKey);
+            return pluginKey;
+        }
+
+        private string FindPluginName()
+        {
+            string pluginName;
+            this.properties.TryGetValue(WellKnownPluginProperties.PluginName, out pluginName);
+            return pluginName;
+        }
+
+        private void AddCoreSources(string workingDirectory)
+        {
+            SourceGenerator.CreateSourceFiles(typeof(RulesPluginGenerator).Assembly, "SonarQube.Plugins.Resources.Core.", workingDirectory, new Dictionary<string, string>());
+
+            foreach (string sourceFile in Directory.GetFiles(workingDirectory, "*.java", SearchOption.AllDirectories))
+            {
+                this.AddSourceFile(sourceFile);
+            }
+        }
+
+        private void ConfigureSourceFileReplacements()
+        {
+            string pluginKey = this.FindPluginKey();
+            string packageName = CorePluginPackageNameTemplate.Replace(WellKnownSourceCodeTokens.Core_PluginKey, pluginKey);
+            string fullPluginClassName = CorePluginEntryClassTemplate.Replace(WellKnownSourceCodeTokens.Core_PluginPackage, packageName);
+
+            this.sourceFileReplacements.Add(WellKnownSourceCodeTokens.Core_PluginPackage, packageName);
+            this.sourceFileReplacements.Add(WellKnownSourceCodeTokens.Core_PluginKey, pluginKey);
+            this.sourceFileReplacements.Add(WellKnownSourceCodeTokens.Core_PluginName, this.FindPluginName());
+
+            this.SetProperty(WellKnownPluginProperties.Class, fullPluginClassName);
+
+            // Build and set the list of extensions to be exported
+            string javaList = string.Join(", " + Environment.NewLine, this.extensionClasses);
+            this.sourceFileReplacements.Add(ExtensionListToken, javaList);
+        }
+
+        private void AddCoreJars(string workingDirectory)
+        {
+            // Unpack and reference the required jar files
+            SourceGenerator.UnpackReferencedJarFiles(typeof(RulesPluginGenerator).Assembly, "SonarQube.Plugins.Resources.Core.", workingDirectory);
+            foreach (string jarFile in Directory.GetFiles(workingDirectory, "*.jar"))
+            {
+                this.AddReferencedJar(jarFile);
+            }
+        }
+
+        /// <summary>
+        /// Perform token-based substitution into the supplied source files prior to compiling
+        /// </summary>
+        private void FixUpSourceFiles()
```

Comment:
Consider using "transform" or transformTokens instead of fixup which is somewhat too generic